packages/mermaid/src/diagrams/git/gitGraphAst.ts

```diff
@@ -0,0 +1,519 @@
+import { log } from '../../logger.js';
+import { cleanAndMerge, random } from '../../utils.js';
+import { getConfig as commonGetConfig } from '../../config.js';
+import common from '../common/common.js';
+import {
+  setAccTitle,
+  getAccTitle,
+  getAccDescription,
+  setAccDescription,
+  clear as commonClear,
+  setDiagramTitle,
+  getDiagramTitle,
+} from '../common/commonDb.js';
+import type {
+  DiagramOrientation,
+  Commit,
+  GitGraphDB,
+  CommitDB,
+  MergeDB,
+  BranchDB,
+  CherryPickDB,
+} from './gitGraphTypes.js';
+import { commitType } from './gitGraphTypes.js';
+import { ImperativeState } from '../../utils/imperativeState.js';
+
+import DEFAULT_CONFIG from '../../defaultConfig.js';
+
+import type { GitGraphDiagramConfig } from '../../config.type.js';
+interface GitGraphState {
+  commits: Map<string, Commit>;
+  head: Commit | null;
+  branchConfig: Map<string, { name: string; order: number | undefined }>;
+  branches: Map<string, string | null>;
+  currBranch: string;
+  direction: DiagramOrientation;
+  seq: number;
+  options: any;
+}
+
+const DEFAULT_GITGRAPH_CONFIG: Required<GitGraphDiagramConfig> = DEFAULT_CONFIG.gitGraph;
+const getConfig = (): Required<GitGraphDiagramConfig> => {
+  const config = cleanAndMerge({
+    ...DEFAULT_GITGRAPH_CONFIG,
+    ...commonGetConfig().gitGraph,
+  });
+  return config;
+};
+const config = getConfig();
+const mainBranchName = config.mainBranchName;
+const mainBranchOrder = config.mainBranchOrder;
+
+const state = new ImperativeState<GitGraphState>(() => ({
+  commits: new Map(),
+  head: null,
+  branchConfig: new Map([[mainBranchName, { name: mainBranchName, order: mainBranchOrder }]]),
+  branches: new Map([[mainBranchName, null]]),
+  currBranch: mainBranchName,
+  direction: 'LR',
+  seq: 0,
+  options: {},
+}));
```

Comment:
There is still an issue here, but pretty sure it was already present. So this can be fixed in a separate PR if required.

- User passes mainBranchName in config in initialize, it will be set in line 49.
- If the user changes that later, or passes a different value in yaml, it won't change what's there in 49. 

I'd suggest moving all three of these into the imperative state, so they'll be reset after each diagram. 

![image](https://github.com/user-attachments/assets/73c8155c-7da0-4a11-854b-fdc508bee95c)